hooks/tk-multi-loader2/basic/scene_actions.py

```diff
@@ -147,26 +149,54 @@ def execute_action(self, name, params, sg_publish_data):
         app.log_debug("Execute action called for action %s. "
                       "Parameters: %s. Publish Data: %s" % (name, params, sg_publish_data))
 
-        # resolve path
-        # toolkit uses utf-8 encoded strings internally and the After Effects API expects unicode
-        # so convert the path to ensure filenames containing complex characters are supported
-        path = self.get_publish_path(sg_publish_data).decode('utf-8')
-
-        if self.parent.engine.is_adobe_sequence(path):
-            frame_range = self.parent.engine.find_sequence_range(path)
-            if frame_range:
-                glob_path = re.sub("[\[]?([#@]+|%0\d+d)[\]]?", "*{}".format(frame_range[0]), path)
-                for each_path in sorted(glob.glob(glob_path)):
-                    path = each_path
-                    break
-
-        if not os.path.exists(path):
-            raise IOError("File not found on disk - '%s'" % path)
-
-        if name == _ADD_TO_COMP:
-            self._add_to_comp(path)
-        if name == _ADD_TO_PROJECT:
-            self.parent.engine.import_filepath(path)
+        # If the loader2 dialog is closed during processing it causes some stability problems
+        # on Windows and OSX. We'll just not allow it to be closed while we're working.
+        top_level_widgets = [w for w in QtGui.QApplication.topLevelWidgets() if isinstance(w, TankQDialog)]
```

Comment:
I'm worried here that we're using a class that is part of the internal API of Toolkit. It's unlikely to change, but it's still not part of the official interface of tk-core, which means that accessing it makes it official and we'll have to support this down the road.

From my understanding, during the meeting today you said that this was both an AE and a Photoshop problem. It's also very likely a Unity problem as well. So it's starting to look like a pattern. Maybe this fix should go in the loader app instead. In there we can easily call self.window() and edit the window flags for the current window only. If this sort of issue becomes more common place in other Toolkit apps then we can start thinking of another way to fix this and at least we won't have introduced a dependency on TankQDialog.

What do you think?